### PR TITLE
refactor: refactor main function, add lib.rs and refactor config/cli related code

### PR DESCRIPTION
**Pull request by fabriziosestito**
## Description

This PR splits policy server code in lib/bin by adding a `lib.rs` file.
In this way we can call the `policy_server::run` function passing the `Config` struct just introduced by this PR from integration tests in the `tests` folder.

Related to: https://github.com/orgs/kubewarden/projects/6?pane=issue&itemId=44457602

### CODE REVIEW

**Review by flavio — APPROVED** [review]
Thanks for the cleanup, this looks much better

**Review by viccuad — APPROVED** [review]
LGTM!

File: Cargo.lock [hunk 1]
```diff
@@ -3734,6 +3734,7 @@ dependencies = [
  "percent-encoding",
  "pin-project-lite",
  "rustls",
+ "rustls-native-certs",
```
Comment (viccuad) [inline Cargo.lock:3737]: I wonder why this addition, it shouldn't be needed. I suppose cargo will drop it later on.

Comment (viccuad) [reply Cargo.lock:3736]: Tried a cargo check of the branch locally, and it gets removed. Not an issue though!

**Pull request merged**